### PR TITLE
Modify EventPipe runtime event generator to emit NULL string correctly

### LINKED ISSUE #43089: MethodLoadVervose_V1 received with incorrect layout via EventPipe
When I'm using EventPipe to collect events from .NET 5 application built for x86 on Windows, sometimes at the beginning of the session I receive a bunch of `MethodLoadVerbose_V1` events (id 143, version 1) with the invalid layout. Maybe this issue can be reproduced on other configurations too.

These events are based on `MethodLoadUnloadVerbose_V1` template and should have following structure:
```
<data name="MethodID" inType="win:UInt64" outType="win:HexInt64" />
<data name="ModuleID" inType="win:UInt64" outType="win:HexInt64" />
<data name="MethodStartAddress" inType="win:UInt64" outType="win:HexInt64" />
<data name="MethodSize" inType="win:UInt32" outType="win:HexInt32" />
<data name="MethodToken" inType="win:UInt32" outType="win:HexInt32" />
<data name="MethodFlags" inType="win:UInt32" map="MethodFlagsMap" />
<data name="MethodNamespace" inType="win:UnicodeString" />
<data name="MethodName" inType="win:UnicodeString" />
<data name="MethodSignature" inType="win:UnicodeString" />
<data name="ClrInstanceID" inType="win:UInt16" />
```

but I receive events with only one string (`MethodName` actually).  Examples of such events are
```
[2020-10-05 19:33:08.903096]: <debug> Memory dump:
  0000  cc 30 43 03 00 00 00 00 00 00 00 00 00 00 00 00  .0C.............
  0010  cc 30 43 03 00 00 00 00 34 00 00 00 00 00 00 00  .0C.....4.......
  0020  10 00 00 00 40 00 4e 00 65 00 77 00 4f 00 62 00  ....@.N.e.w.O.b.
  0030  6a 00 65 00 63 00 74 00 00 00 0b 00              j.e.c.t.....
[2020-10-05 19:33:08.903096]: <debug> Memory dump:
  0000  00 31 43 03 00 00 00 00 00 00 00 00 00 00 00 00  .1C.............
  0010  00 31 43 03 00 00 00 00 54 00 00 00 00 00 00 00  .1C.....T.......
  0020  10 00 00 00 40 00 4e 00 65 00 77 00 4f 00 62 00  ....@.N.e.w.O.b.
  0030  6a 00 65 00 63 00 74 00 41 00 6c 00 69 00 67 00  j.e.c.t.A.l.i.g.
  0040  6e 00 38 00 00 00 0b 00                          n.8.....
[2020-10-05 19:33:08.903096]: <debug> Memory dump:
  0000  54 31 43 03 00 00 00 00 00 00 00 00 00 00 00 00  T1C.............
  0010  54 31 43 03 00 00 00 00 74 00 00 00 00 00 00 00  T1C.....t.......
  0020  10 00 00 00 40 00 42 00 6f 00 78 00 00 00 0b 00  ....@.B.o.x.....
[2020-10-05 19:33:08.903096]: <debug> Memory dump:
  0000  c8 31 43 03 00 00 00 00 00 00 00 00 00 00 00 00  .1C.............
  0010  c8 31 43 03 00 00 00 00 4c 00 00 00 00 00 00 00  .1C.....L.......
  0020  10 00 00 00 40 00 4e 00 65 00 77 00 41 00 72 00  ....@.N.e.w.A.r.
  0030  72 00 61 00 79 00 31 00 4f 00 62 00 6a 00 65 00  r.a.y.1.O.b.j.e.
  0040  63 00 74 00 00 00 0b 00                          c.t.....
```

As far as I understand, these events are generated by `ETW::MethodLog::SendHelperEvent` from `src/coreclr/src/vm/eventtrace.cpp`:
```
VOID ETW::MethodLog::SendHelperEvent(ULONGLONG ullHelperStartAddress, ULONG ulHelperSize, LPCWSTR pHelperName)
{
    WRAPPER_NO_CONTRACT;
    if(pHelperName)
    {
         PCWSTR szDtraceOutput1=W("");
         ULONG methodFlags = ETW::MethodLog::MethodStructs::JitHelperMethod; // helper flag set
         FireEtwMethodLoadVerbose_V1(ullHelperStartAddress,
                                     0,
                                     ullHelperStartAddress,
                                     ulHelperSize,
                                     0,
                                     methodFlags,
                                     NULL,
                                     pHelperName,
                                     NULL,
                                     GetClrInstanceId());
    }
}
```

And the issue is caused by using `NULL`s here instead of empty strings. I don't found an implementation of `FireEtwMethodLoadVerbose_V1` (is it generated?). If expected using of empty strings instead of `NULL`s in `ETW::MethodLog::SendHelperEvent` can cause issues in other subsystems, the issue can be fixed with additional checks in the `FireEtwMethodLoadVerbose_V1`.

As far as I understand, this issue relates to the issue #8008.

### PR DESCRIPTION
**Pull request by sywhang**
Fix #43089 

EventPipe-generated events whose parameters are string just write nothing if it has the value NULL. ETW events on the contrary will write "NULL". The fix changes EventPipe to write `NULL` string like ETW does, not only because that's parity with ETW but also that is what the Manifest describes the event and TraceEvent can't correctly parse the event if we don't write anything there.

### CODE REVIEW

File: src/coreclr/scripts/genEventPipe.py [hunk 1]
```diff
@@ -178,9 +178,19 @@ def generateWriteEventBody(template, providerName, eventName):
 
     bool success = true;
 """ % (template.estimated_size, template.estimated_size)
-
     fnSig = template.signature
     pack_list = []
+
+    # Write out replacement for any instasnces of NULL for UnicodeString type input.
+    # ETW translates these to "NONE", so we are doing the same for EventPipe.
```
Comment (jkotas) [inline src/coreclr/scripts/genEventPipe.py:185]: The comment says NONE, but the code below uses NULL. Is it right?

File: src/coreclr/scripts/genEventPipe.py [hunk 2]
```diff
@@ -178,9 +178,19 @@ def generateWriteEventBody(template, providerName, eventName):
 
     bool success = true;
 """ % (template.estimated_size, template.estimated_size)
-
     fnSig = template.signature
     pack_list = []
+
+    # Write out replacement for any instasnces of NULL for UnicodeString type input.
```
Comment (jkotas) [inline src/coreclr/scripts/genEventPipe.py:184]:
```suggestion
    # Write out replacement for any instances of NULL for UnicodeString type input.
```

File: src/coreclr/scripts/genEventPipe.py (hunk 1 above)
Comment (sywhang) [reply src/coreclr/scripts/genEventPipe.py:185]: oops - yes it should be "NULL" : )

**Review by josalem — APPROVED** [review]

**Review by stephentoub — APPROVED** [review]

**Review by davmason — APPROVED** [review]

**Comment by davmason** [comment]
@sywhang not sure if you saw or not, but this is showing up on the stale PR reports now

**Comment by sywhang** [comment]
@davmason Thanks, there was a test failure that I had to address.  I'll merge after I see the CI pass.

**Pull request merged**